src/components/ContentNode/LinkableHeading.vue

```diff
@@ -0,0 +1,84 @@
+<!--
+  This source file is part of the Swift.org open source project
+
+  Copyright (c) 2022 Apple Inc. and the Swift project authors
+  Licensed under Apache License v2.0 with Runtime Library Exception
+
+  See https://swift.org/LICENSE.txt for license information
+  See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+
+<template>
+  <component
+   :id="anchor"
+   :is="`h${level}`"
+  >
+    <router-link
+      v-if="anchor && !isTargetIDE"
+      :to="{ hash: `#${anchor}` }"
+      class="header-anchor"
+      aria-label="Scroll to section"
+      @click="handleFocusAndScroll(anchor)"
+    >
+      <slot />
+      <LinkIcon class="icon" aria-hidden="true"/>
+    </router-link>
+    <template v-else>
+      <slot />
+    </template>
+  </component>
+</template>
+
+<script>
+import scrollToElement from 'docc-render/mixins/scrollToElement';
+import LinkIcon from 'theme/components/Icons/LinkIcon.vue';
+
+export default {
+  name: 'LinkableHeading',
+  mixins: [scrollToElement],
+  components: {
+    LinkIcon,
+  },
+  props: {
+    anchor: {
+      type: String,
+      required: false,
+    },
+    level: {
+      type: Number,
+      default: () => 2,
+      validator: v => v >= 1 && v <= 6,
+    },
+  },
+  inject: {
+    isTargetIDE: {
+      default: () => false,
+    },
+  },
+};
+</script>
+
+<style scoped lang="scss">
+@import 'docc-render/styles/_core.scss';
+$icon-margin: 7px;
+
+.header-anchor {
+  color: var(--colors-text, var(--color-text));
```

Comment:
I think we should potentially remove this since these headings are currently being assigned the `--colors-header-text` value in a global/base h1-h6 rule, and this could potentially change the API, even though it's not noticeable at the moment since these colors are probably the same by default.

If removing it isn't sufficient, we should consider changing it to use `--colors-header-text`, but I would recommend just removing this line if possible to make it as close as possible to what it was before.

Reply:
This should do it :) It will override the `color` for anchors and inherit the heading color.

```suggestion
  color: inherit;
```

Reply:
I applied `color: inherit;`, good idea @dobromir-hristov !

Reply:
and thanks @mportiz08 for pointing it out!